<?xml version="1.0"?>
<launch>

  <arg name="sim"		default="false"/>
  <arg name="traj"		default="true"/>
  <arg name="speed"		default="13.0"/>
  <arg name="turn"		default="13.0"/>
  <arg name="repeat_rate"	default="100.0"/>

  <!-- Spawn teleop_twist_keyboard -->
  <node name="teleop_twist_keyboard"
	pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py"
	output="screen">
    <param name="speed"		value="$(arg speed)"/>
    <param name="turn"		value="$(arg turn)"/>
    <param name="repeat_rate"	value="$(arg repeat_rate)"/>
    <param name="stamped"	value="true"/>
    <param name="frame_id"	value="arm_base_link"/>
    <remap from="/cmd_vel"	to="/servo_server/delta_twist_cmds"/>
  </node>

  <!-- Spawn servo_server -->
  <include file="$(dirname)/khi_servo_server.launch">
    <arg name="server"		value="servo_server"/>
    <arg name="sim"		value="$(arg sim)"/>
    <arg name="unitless"	value="false"/>
    <arg name="traj"		value="$(arg traj)"/>
  </include>

</launch>
